superset/tasks/schedules.py

```diff
@@ -551,7 +551,12 @@ def schedule_alert_query(  # pylint: disable=unused-argument
         return
 
     if report_type == ScheduleType.alert:
-        if run_alert_query(schedule, dbsession):
+        # recipients is only specified when task is called from view for a test email
```

Comment:
add additional flag to determine if that is a test run e.g. you already have a flag in the form